### PR TITLE
Fix toBytes() with protobuf codec

### PR DESCRIPTION
**Pull request by bclermont**

### CODE REVIEW

File: codec/protobuf/simple_user.pb.go [hunk 1]
```diff
@@ -27,51 +27,27 @@ var _ = math.Inf
 const _ = proto.ProtoPackageIsVersion1
 
 type SimpleUser struct {
-	Id               *uint64 `protobuf:"varint,1,req,name=id" json:"id,omitempty"`
-	Name             *string `protobuf:"bytes,2,req,name=name" json:"name,omitempty"`
-	Age              *int32  `protobuf:"varint,3,opt,name=age,def=0" json:"age,omitempty"`
-	XXX_unrecognized []byte  `json:"-"`
+	Id   uint64 `protobuf:"varint,1,opt,name=id" json:"id,omitempty" storm:"id"`
```
Comment (bclermont) [inline codec/protobuf/simple_user.pb.go]: blame `protoc` this is generated code :P

File: codec/protobuf/protobuf.go [hunk 2]
```diff
@@ -18,7 +19,8 @@ type protobufCodec int
 func (c protobufCodec) Encode(v interface{}) ([]byte, error) {
```
Comment (bep) [inline codec/protobuf/protobuf.go]: I agree about the fix here, but in this case the implementation differs from the interface definition (it is not a pure protobuf encode  anymore), so a proper GoDoc on the Encode method would be nice.

Comment (bclermont) [inline codec/protobuf/protobuf.go]: done

File: codec/protobuf/protobuf_test.go [hunk 3]
```diff
@@ -2,18 +2,27 @@ package protobuf
...
-	assert.True(t, u1.GetId() == u2.GetId())
+	assert.True(t, u1.Id == u2.Id)
+}
+
+func TestSave(t *testing.T) {
+	dir, _ := ioutil.TempDir(os.TempDir(), "storm")
+	defer os.RemoveAll(dir)
+	db, _ := storm.Open(filepath.Join(dir, "storm.db"), storm.Codec(Codec))
+	u := SimpleUser{Id: 1, Name: "John"}
+	err := db.Save(&u)
```
Comment (bep) [inline codec/protobuf/protobuf_test.go]: For this change I would also read the user back from the db and make sure they are the same.

Comment (bclermont) [inline codec/protobuf/protobuf_test.go]: test added

**Comment by asdine** [comment]
Thanks!

**Pull request merged**